query/src/main/scala/filodb/query/LogicalPlan.scala

```diff
@@ -35,6 +37,11 @@ case class RawSeries(rangeSelector: RangeSelector,
                      filters: Seq[ColumnFilter],
                      columns: Seq[String]) extends RawSeriesPlan
 
+
+case class Metadata(rawSeries: RawSeries,
```

Comment:
Don't you need two separate LogicalPlans?

    IndexValuesMetadata
    PartitionKeysMetadata

Each of those has separate definitions and should not refer to RawSeries, which implies getting data from raw series.